packages/shopping/src/middlewares/shoppy.middleware.ts

```diff
@@ -0,0 +1,47 @@
+import {Middleware, MiddlewareContext, RestBindings} from '@loopback/rest';
+import {Next} from '@loopback/context';
+import {
+  AUTHENTICATION_STRATEGY_NOT_FOUND,
+  USER_PROFILE_NOT_FOUND,
+} from '@loopback/authentication';
+
+export const ShoppyMiddleware: Middleware = async (
+  ctx: MiddlewareContext,
+  next: Next,
+) => {
+  const send = await ctx.get(RestBindings.SequenceActions.SEND);
+  const reject = await ctx.get(RestBindings.SequenceActions.REJECT);
+  try {
+    const result = await next();
+    send(ctx.response, result);
+  } catch (error) {
+    //
+    // The authentication action utilizes a strategy resolver to find
+    // an authentication strategy by name, and then it calls
+    // strategy.authenticate(request).
+    //
+    // The strategy resolver throws a non-http error if it cannot
+    // resolve the strategy. When the strategy resolver obtains
+    // a strategy, it calls strategy.authenticate(request) which
+    // is expected to return a user profile. If the user profile
+    // is undefined, then it throws a non-http error.
+    //
+    // It is necessary to catch these errors and add HTTP-specific status
+    // code property.
+    //
+    // Errors thrown by the strategy implementations already come
+    // with statusCode set.
+    //
+    // In the future, we want to improve `@loopback/rest` to provide
+    // an extension point allowing `@loopback/authentication` to contribute
+    // mappings from error codes to HTTP status codes, so that application
+    // don't have to map codes themselves.
+    if (
+      error.code === AUTHENTICATION_STRATEGY_NOT_FOUND ||
+      error.code === USER_PROFILE_NOT_FOUND
+    ) {
+      Object.assign(error, {statusCode: 401 /* Unauthorized */});
+    }
+    reject(ctx, error);
+  }
```

Comment:
Any reason for this middleware? AFAIK, this is already handled by the Authentication middleware:

https://github.com/strongloop/loopback-next/blob/883e9b31b0e5281085a0c9f091f648e84fef795e/packages/authentication/src/providers/auth-action.provider.ts#L130-L141

Reply:
Good catch. To be honest am looking at @raymondfeng or anyone who has more exposure to middleware to weigh in. One of the reasons why I did this change I feel like for most users its still a grey area myself included

Reply:
If you want to customize the error mapping, please check out an example at https://github.com/strongloop/loopback-next/blob/master/examples/validation-app/src/middleware/validation-error.middleware.ts.

Reply:
Thanks @raymondfeng I will apply it on my projects and update PR accordingly

Reply:
Didn't find a use case for the middleware provider on this repo. However, it did help me on my projects at work. I will remove the customisation for now until i find a way i can illustrate this better for our users